src/server/socket.js

```diff
@@ -309,6 +309,7 @@ SocketServer.prototype._subscribeTo = function (host, msg, handler, once) {
     var socket = this._hostSockets[host],
         method = once ? 'once' : 'on';
     if (socket) {
+        socket.removeListener(msg, handler);
```

Comment:
Talked it over with @MSLaguana, and agree this looks fine to me. Couple of things:

* Would still be good if you could take a look @busykai.
* I'm concerned this is just a symptom of a bigger problem. It would be good if the three of us could get together and talk this through at some point (perhaps @MSLaguana you could send an email describing the scenario in which this occurs).